mitodl/mit-learn
# add a trailing slash to non contentfile OCW resources

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/6187

### Description (What does it do?)
This PR alters the OCW ETL code to add a trailing slash to the end of OCW URLs that reference pages or whole sites. OCW sites are generated with the Hugo static site generator and are configured to always generate a folder with an `index.html` file inside of it for each page. Although this PR is about indexing in OER commons and duplicates appearing, this is important to consider because not all OCW content are pages. Some of the content files are raw images, pdf's, etc. These URLs have not been modified, because a slash at the end of them would be incorrect.

### How can this be tested?
 - Ensure that you have proper AWS credentials in your environment and that you have set `OCW_LIVE_BUCKET=ocw-content-live-production`
 - Spin up your containers
 - Run `docker compose exec web ./manage.py backpopulate_ocw_data --skip-contentfiles --overwrite`
 - After the backpopulate has completed, enter a Django shell with `docker compose exec web ./manage.py shell`
 - Run the following commands in the shell:
 ```py
from learning_resources.models import *
LearningResource.objects.filter(resource_type="course", offered_by_id="ocw")[0].url
 ```
 - You should see an OCW URL with a trailing slash like `'http://ocw.mit.edu/courses/4-463-building-technology-iii-building-structural-systems-fall-2004/'`
